lib/pluginmanager/package.rb

```diff
@@ -0,0 +1,32 @@
+class LogStash::PluginManager::Package < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "compress package as a tar.gz file", :default => false
+  option "--zip", :flag, "compress package as a zip file", :default => false
+  option "--clean", :flag, "clean up the generated dump of plugins", :default => false
+
+  def execute
+    signal_error("File #{LogStash::Environment::GEMFILE_PATH} does not exist or is not writable, aborting") unless File.writable?(LogStash::Environment::GEMFILE_PATH)
+    puts("packaging plugins for offline usage")
+    LogStash::Bundler.invoke!({:package => true, :all => true})
+    compress_cache(LogStash::Environment::CACHE_PATH)
+    rm_rf(LogStash::Environment::CACHE_PATH) if clean?
+  end
+
+  private
+
+  def compress_cache(location)
+    require "logstash/util/compress"
+    target_file = File.join(LogStash::Environment::LOGSTASH_HOME, "plugins_cache")
```

Comment:
I added a few improvements in here, basically a check for this file plus options to override it from the command line or user interaction. Let me know if this makes sense to you.